src/InviteesModel.cpp

```diff
@@ -85,21 +85,30 @@ InviteesModel::mxids()
     return mxidList;
 }
 
-Invitee::Invitee(QString mxid, QObject *parent)
+Invitee::Invitee(QString mxid, QString displayName, QString avatarUrl, QObject *parent)
   : QObject{parent}
   , mxid_{std::move(mxid)}
 {
-    http::client()->get_profile(
-      mxid_.toStdString(), [this](const mtx::responses::Profile &res, mtx::http::RequestErr err) {
-          if (err) {
-              nhlog::net()->warn("failed to retrieve profile info");
-              emit userInfoLoaded();
-              return;
-          }
-
-          displayName_ = QString::fromStdString(res.display_name);
-          avatarUrl_   = QString::fromStdString(res.avatar_url);
+    // checking for empty avatarUrl will cause profiles to be loaded that don't have an avatar
+    // to needlessly be loaded. Can we make sure we either provide both or none?
```

Comment:
Do we need to? We can just always load it, can't we?

Reply:
I guess that is fine, yes. Note that some users have an avatar, but no displayname. Maybe you want to pass the mxid in that case or check also for the avatarUrl here?

Reply:
I didn't consider the case of an empty displayname. Possibly rather than passing empty strings, we need to specify whether user info should be loaded or not, since we know that. I just decided to use what's there and not change function declarations.

Is it OK to assume that if both avatarUrl and displayName are empty, we need to fetch a profile? If only one is empty, we can assume that profile info is complete, if both are empty we may either not have passed them and thus need to fetch the profile or the user has indeed empty displayname and avatarUrl, but fetching the profile isn't going to break anything, it's just an unnecessary request.